packager/media/file/udp_file.cc

```diff
@@ -90,58 +94,93 @@ bool UdpFile::Tell(uint64_t* position) {
   return false;
 }
 
+#if defined(OS_WIN)
+class LibWinsockInitializer {
+ public:
+  LibWinsockInitializer() {
+    WSADATA wsa_data;
+    error_ = WSAStartup(MAKEWORD(2, 2), &wsa_data);
+  }
+
+  ~LibWinsockInitializer() {
+    if (error_ == 0) WSACleanup();
+  }
+
+  int error() const { return error_; }
+
+ private:
+  int error_;
+};
+#endif  // defined(OS_WIN)
+
 class ScopedSocket {
  public:
-  explicit ScopedSocket(int sock_fd)
+  explicit ScopedSocket(SOCKET sock_fd)
       : sock_fd_(sock_fd) {}
 
   ~ScopedSocket() {
-    if (sock_fd_ != kInvalidSocket)
+    if (sock_fd_ != INVALID_SOCKET)
       close(sock_fd_);
   }
 
-  int get() { return sock_fd_; }
+  SOCKET get() { return sock_fd_; }
 
-  int release() {
-    int socket = sock_fd_;
-    sock_fd_ = kInvalidSocket;
+  SOCKET release() {
+    SOCKET socket = sock_fd_;
+    sock_fd_ = INVALID_SOCKET;
     return socket;
   }
 
  private:
-  int sock_fd_;
+  SOCKET sock_fd_;
 
   DISALLOW_COPY_AND_ASSIGN(ScopedSocket);
 };
 
 bool UdpFile::Open() {
-  DCHECK_EQ(kInvalidSocket, socket_);
+#if defined(OS_WIN)
+  static LibWinsockInitializer lib_winsock_initializer;
+  if (lib_winsock_initializer.error() != 0) {
+    LOG(ERROR) << "Winsock start up failed with error "
+               << lib_winsock_initializer.error();
+    return false;
+  }
+#endif  // defined(OS_WIN)
+
+  DCHECK_EQ(INVALID_SOCKET, socket_);
 
   std::unique_ptr<UdpOptions> options =
       UdpOptions::ParseFromString(file_name());
   if (!options)
     return false;
 
   ScopedSocket new_socket(socket(AF_INET, SOCK_DGRAM, 0));
-  if (new_socket.get() == kInvalidSocket) {
+  if (new_socket.get() == INVALID_SOCKET) {
     LOG(ERROR) << "Could not allocate socket.";
     return false;
   }
 
-  struct sockaddr_in local_sock_addr;
-  bzero(&local_sock_addr, sizeof(local_sock_addr));
-  // TODO(kqyang): Support IPv6.
-  local_sock_addr.sin_family = AF_INET;
-  local_sock_addr.sin_port = htons(options->port());
+  struct in_addr local_in_addr = {0};
   if (inet_pton(AF_INET, options->address().c_str(),
-                &local_sock_addr.sin_addr) != 1) {
+                &local_in_addr) != 1) {
     LOG(ERROR) << "Malformed IPv4 address " << options->address();
     return false;
   }
 
+  struct sockaddr_in local_sock_addr = {0};
+  // TODO(kqyang): Support IPv6.
+  local_sock_addr.sin_family = AF_INET;
+  local_sock_addr.sin_port = htons(options->port());
+  if (IsIpv4MulticastAddress(local_in_addr)) {
```

Comment:
Can you assign this to a local variable to avoid duplicate computation below in line 198, e.g.

const bool is_multicast = IsIpv4MulticastAddress(local_in_addr);
...